### PR TITLE
Upgrade spring cloud azure module to 1.0.0.M2.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/resources/templates/azure-eventhub-binder/SinkExample.java [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
-    public void handleMessage(String message) {
-        LOGGER.info("New message received: " + message);
+    public void handleMessage(String message, @Header(AzureHeaders.CHECKPOINTER) Checkpointer checkpointer) {
+        LOGGER.info("New message received: '{}'", message);
+        checkpointer.success().handle((r, ex) -> {
+            if (ex == null) {
+                LOGGER.info("Message '{}' successfully checkpointed", message);
+            }
+
+            return null;
```
Comment (sophiaso) [inline src/main/resources/templates/azure-eventhub-binder/SinkExample.java]: Do we need to log the exception?

Comment (warrenzhu25) [reply src/main/resources/templates/azure-eventhub-binder/SinkExample.java]: Yes, should log exception either.

File: src/main/resources/templates/azure-cosmosdb/CosmosdbController.java [hunk 2]
```diff
@@ -10,11 +10,14 @@
...
 public class CosmosdbController {
 
+    private static final User USER = new User("id", "first-name", "last-name", "address");
+
     @Autowired
     private UserRepository repository;
 
     @GetMapping(value = "/users")
     public List<User> getUsers() {
+        this.repository.save(USER);
```
Comment (sophiaso) [inline src/main/resources/templates/azure-cosmosdb/CosmosdbController.java]: Save data for every get request? Move initialize to constructor, and inject repository by constructor

File: src/main/resources/templates/azure-eventhub-binder/SinkExample.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/resources/templates/azure-eventhub-binder/SinkExample.java]: OK. I will update this.

File: src/main/resources/templates/azure-cosmosdb/CosmosdbController.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/resources/templates/azure-cosmosdb/CosmosdbController.java]: Roger that.

File: src/main/resources/templates/azure-eventhub-binder/SinkExample.java (hunk 1 above)
Comment (warrenzhu25) [reply src/main/resources/templates/azure-eventhub-binder/SinkExample.java]: We can keep this same as spring cloud sample for now.

Comment (Incarnation-p-lee) [reply src/main/resources/templates/azure-eventhub-binder/SinkExample.java]: OK.

**Pull request merged**